overlore/sqlite/townhall_db.py

```diff
@@ -0,0 +1,186 @@
+from __future__ import annotations
+
+import json
+import logging
+from datetime import datetime
+from sqlite3 import Connection
+from typing import Any, cast
+
+import sqlite_vss
+
+from overlore.sqlite.base_db import BaseDatabase
+from overlore.sqlite.errors import CosineSimilarityNotFoundError
+from overlore.sqlite.types import StoredVector
+
+logger = logging.getLogger("overlore")
+
+
+class TownhallDatabase(BaseDatabase):
+    _instance: TownhallDatabase | None = None
+    EXTENSIONS: list[str] = []
+    FIRST_BOOT_QUERIES: list[str] = [
+        """
+            CREATE TABLE IF NOT EXISTS townhall (
+                discussion TEXT,
+                townhall_input TEXT,
+                realm_id INTEGER NOT NULL,
+                ts TEXT
+            );
+        """,
+        """
+            CREATE TABLE IF NOT EXISTS daily_townhall_tracker (
+                realm_id INTEGER NOT NULL,
+                event_row_id TEXT
+            );
+        """,
+        """
+            CREATE TABLE IF NOT EXISTS realm_plotline (
+                plotline TEXT,
+                realm_id INTEGER NOT NULL
+            );
+        """,
+        """
+            CREATE TABLE IF NOT EXISTS npc_thought (
+                npc_entity_id INTEGER NOT NULL,
+                thought TEXT
+            );
+        """,
+        """
+            CREATE VIRTUAL TABLE IF NOT EXISTS vss_npc_thought using vss0(
+                embedding(1536)
+            );
+        """,
+    ]
+
+    @classmethod
+    def instance(cls) -> TownhallDatabase:
+        if cls._instance is None:
+            logger.debug("Creating townhall db interface")
+            cls._instance = cls.__new__(cls)
+        return cls._instance
+
+    def __init__(self) -> None:
+        raise RuntimeError("Call instance() instead")
+
+    def _preload(self, db: Connection) -> None:
+        sqlite_vss.load(db)
+
+    def init(self, path: str = "./databases/townhall.db") -> TownhallDatabase:
+        self._init(path, self.EXTENSIONS, self.FIRST_BOOT_QUERIES, [], self._preload)
+        return self
+
+    # def query_nearest_neighbour(self, query_embedding: str, npc_entity_id: int, limit: int = 1) -> list[StoredVector]:
+    #     if limit <= 0:
+    #         raise ValueError("Limit must be higher than 0")
+    #     if self.get_entries_count() == (0, 0):
+    #         return []
+
+    #     query = """
+    #         SELECT v.rowid, v.distance FROM vss_npc_thought v
+    #         INNER JOIN npc_thought t ON v.rowid = t.rowid
+    #         WHERE t.npc_entity_id = ? AND vss_search(embedding, vss_search_params(?, ?))
+    #     """
+
+    #     values = (npc_entity_id, json.dumps(query_embedding), limit + 1)
+
+    #     return self.execute_query(query, values)
```

Comment:
Remove, if you need this code in the future you can still go back in previous commits